packages/core/addon/services/fragment-factory.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import Service, { inject as service } from '@ember/service';
+import Store from 'ember-data/store';
+import Column from 'navi-data/models/metadata/column';
+import ColumnFragment from '../models/bard-request-v2/fragments/column';
+import FilterFragment from '../models/bard-request-v2/fragments/filter';
+import SortFragment from '../models/bard-request-v2/fragments/sort';
+import { dasherize } from '@ember/string';
+
+interface StoreWithFragment extends Store {
+  createFragment(fragmentName: string, attributes: object): ColumnFragment | FilterFragment | SortFragment;
+}
+export default class FragmentFactory extends Service {
+  @service store!: StoreWithFragment;
+
+  /**
+   * Builds a request v2 column fragment given meta data object.
+   * @param meta - The metadata object to use as field to build this fragment from
+   * @param parameters - parameters to attach to column, if none pass empty object `{}`
+   * @param alias - optional alias for this column
+   * @param dimensionField - dimension field if passing in a dimension `id` or `description` for example
+   */
+  createColumnFromMeta(
+    columnMetadata: Column,
+    parameters: Dict<string> = {},
+    alias?: string,
+    dimensionField?: string
+  ): ColumnFragment {
+    const type = this._getMetaColumnType(columnMetadata);
+    const column = this.store.createFragment('bard-request-v2/fragments/column', {
+      field: dimensionField ? `${columnMetadata.id}.${dimensionField}` : columnMetadata.id,
+      type,
+      parameters,
+      alias
+    }) as ColumnFragment;
+    column.source = columnMetadata.source;
+    return column;
```

Comment:
```suggestion
    const field = dimensionField ? `${columnMetadata.id}.${dimensionField}` : columnMetadata.id;
    return this.createColumn(type, columnMetadata.source, field, parameters, alias);
```

Could follow this pattern for sort and filter as well